### PR TITLE
880: columns visibilty modal behaviour

### LINKED ISSUE #880: Columns visibility modal don't work
fix Columns visibility modal in score

### PR DESCRIPTION
**Pull request by valerydluski**
Fixes #880

### CODE REVIEW

File: client/src/modules/Score/components/ScoreTable/index.tsx [hunk 1]
```diff
@@ -84,17 +88,19 @@ export function ScoreTable(props: Props) {
...
+
   const [notVisibleColumns, setNotVisibleColumns] = useLocalStorage<string[]>('notVisibleColumns', []);
 
   const handleModalCancel = () => {
     setIsVisibleSettings(!isVisibleSetting);
   };
 
   const handleModalOk = (values: Store) => {
-    setNotVisibleColumns(Object.keys(values).filter((value: string) => values[value] === false));
+    setNotVisibleColumns(Object.keys(values).filter((value: string) => !values[value]));
```
Comment (Pulya10c) [inline client/src/modules/Score/components/ScoreTable/index.tsx:103]: The `!values[value]` operation takes longer than `values [value] === false`. What is the point of replacing it?

**Review by Pulya10c — APPROVED** [review]

File: client/src/modules/Score/components/ScoreTable/index.tsx [hunk 2]
```diff
@@ -65,8 +65,12 @@ export function ScoreTable(props: Props) {
...
         courseService.getCourseTasks(),
       ]);
-      const sortedTasks = courseTasks.filter(task => !!task.studentEndDate || props.course.completed);
-
+      const sortedTasks = courseTasks
+        .filter(task => !!task.studentEndDate || props.course.completed)
+        .map(task => {
+          task.isVisible = !notVisibleColumns?.includes(task.name);
+          return task;
+        });
```
Comment (mikhama) [inline client/src/modules/Score/components/ScoreTable/index.tsx:73]: Hope you're know what are you doing.
If there weren't be filter then you would mutate original `courseTasks` array.

File: client/src/modules/Score/components/ScoreTable/index.tsx (hunk 1 above)
Comment (valerydluski) [reply client/src/modules/Score/components/ScoreTable/index.tsx]: done

File: client/src/modules/Score/components/ScoreTable/index.tsx (hunk 2 above)
Comment (valerydluski) [reply client/src/modules/Score/components/ScoreTable/index.tsx]: done

File: client/src/modules/Score/components/ScoreTable/index.tsx [hunk 3]
```diff
@@ -84,9 +88,11 @@ export function ScoreTable(props: Props) {
     cityName,
     mentor,
     handleSettings: () => setIsVisibleSettings(true),
-    taskColumns: getTaskColumns(courseTasks).filter(column => !notVisibleColumns?.includes(column.name)),
+    taskColumns: getTaskColumns(courseTasks),
   });
 
+  const getVisibleColumns = (columns: any[]) => columns.filter(column => !notVisibleColumns?.includes(column.name));
```
Comment (apalchys) [inline client/src/modules/Score/components/ScoreTable/index.tsx:94]: I wonder do we really need this as a separate function and execute it 2 times below?

we can do like this:

```typescript
const columns = getColumns({ .... })
    .filter(column => !notVisibleColumns?.includes(column.name))
```

**Review by apalchys — APPROVED** [review]

**Pull request merged**